src/Analyzers/CSharp/Tests/UseCollectionInitializer/UseCollectionInitializerTests.cs

```diff
@@ -322,8 +322,8 @@ class C
     void M()
     {
         var c = [|new|] List<int>();
-        c.Add(1);
-        c.Add(2);
+        [|c.Add(|]1);
+        [|c.Add(|]2);
```

Comment:
The unnecessary diagnostic is reported with `ReportDiagnostic.Default`, so if `IDE0017` is set to error, we'll have the three locations as errors?

It would be good to have a test that asserts the additional diagnostics are *always* hidden, even if IDE0017 is set to error.

Reply:
> Do the user see 3 separate diagnostics? or it's just one diagnostic with one main location and two additional unnecessary locations?

It's just one main diagnostic and other two are additional faded locations. The fading logic in use object initializer and use collection initializer was guarded by two fading options, which were false by default and there was no check box exposed in Tools Option to change these option values. I removed these unused fading options after chatting with @CyrusNajmabadi, and now the fading logic correctly lights up and we see these unnecessary code diagnostics.  

> In analyzer code, the unnecessary diagnostic is reported with ReportDiagnostic.Default, so if IDE0017 is set to error, we'll have the three locations as errors?

Yes, that is correct. If we did want the additional faded locations to always be hidden, we'd have to use a separate diagnostic ID for them, which is also reasonable.

I am personally fine with even deleting all the fading related code for use object initializer and use collection initializer analyzers and revert my last commit https://github.com/dotnet/roslyn/pull/62562/commits/27bc7c6a9f7c6dc73ffeda170fc249c6c1c4e06b. This code was basically dead code until now, and I presume we haven't seen users request these additional locations to get faded. @CyrusNajmabadi what do you think?